crates/store/re_types_core/src/reflection.rs

```diff
@@ -285,3 +285,96 @@ impl ArchetypeFieldReflection {
         }
     }
 }
+
+/// An extension of [`ComponentDescriptor`] that helps handling components that are
+/// built into Rerun.
+///
+/// Note that in general, [`ComponentDescriptor::archetype`] does not place any
+/// constraints on the [`ComponentDescriptor::component`] field, which can be arbitrary.
+///
+/// Components that are built into the Rerun viewer follow the convention that
+/// the `component` field starts with the short name of its archetype.
+pub trait ComponentDescriptorExt {
+    /// Returns the field name of a Rerun-builtin type.
+    ///
+    /// This is the result of stripping the Rerun-builtin [`ArchetypeName`]
+    /// from [`ComponentDescriptor::component`].
```

Comment:
but for builtin types we expect the archetype name to be present separately, no? So why would we then go to the `component`? 🤔